syntax = "proto3";

message RecommendationBookRequest {
    int32 user_id = 1;
    int32 max_results = 2;
}

message BookRecommendation {
    int32 id = 1;
    string book_title = 2;
}

message RecommendationBookResponse {
    repeated BookRecommendation recommendations = 1;
}



message BookNameRequest {
    int32 user_id = 1;
	string name = 2;
    int32 max_results = 3;
}




message BookCategoryRequest {
    int32 user_id = 1;
	string category = 2;
    int32 max_results = 3;
}



message BookCategoryResponse {
    repeated BookRecommendation recommendations = 1;
}

message BookNameResponse {
    repeated BookRecommendation recommendations = 1;
}



service RecommendationsBook {
    rpc RecommendBook (RecommendationBookRequest) returns (RecommendationBookResponse);
	rpc SearchByName (BookNameRequest) returns (RecommendationBookResponse);
	rpc SearchByCategory (BookCategoryRequest) returns (RecommendationBookResponse);
}
